src/main/java/com/elastic/support/diagnostics/commands/SystemCallsCmd.java

```diff
@@ -220,29 +196,70 @@ public ProcessBuilder getProcessBuilder() {
 
         // If there was a quote delimited set of parameters to
         // pass it in add it now, sans quote.
-        if(args.length > 1){
+        if (args.length > 1) {
             cmds.add(args[1]);
         }
-
         return cmds;
-
     }
 
+    public String parseEsProcessString(String input, String fileSeparator, String platform) {
 
-    public boolean isJdkPresent(String pathToJavac) {
-        try {
-            File jdk = Paths.get(pathToJavac).toFile();
-            if (jdk.exists()) {
-                return true;
+        String javaExePath = fileSeparator + "bin" + fileSeparator + "java";
+        // Check each line to see if it's an ES java process...
+        if (StringUtils.isEmpty(input)) {
+            return "";
+        }
+        // Windows output is enclosed in quotes
+        if(platform.equals("winOS")){
+            input = input.replace("\"", "");
```

Comment:
There was an issue with spaces in paths - it just doesn't have anything to do with that particular code block. When you're working with the ProcessBuilder all the components such as path and arguments are broken into separate tokens. So it goes in as a unit. In fact, if you leave in the quotes, it won't work. That's in there because I already hit that particular issue.

But there _were_ issues with spaces, and not just for Windows. Spaces in any path the the JDK, whether on Win, Linux, or Mac would break. But that was a function of how I got to the executable String. Tokenizing on whitespace works fine unless there's a path that contains whitespace. As soon as that happens, it breaks.

So I had to revise the parsing process, although the second time through I was able to get all the Windows conditional code out of the the general command.